src/openassets/marker_output.rs

```diff
@@ -56,28 +60,52 @@ impl<D: Decoder> Decodable<D> for Payload{
             loop {
                 let b: u8 = Decodable::consensus_decode(d)?;
                 value |= ((b as u64) & 0x7f) << offset;
-                if (b as u64) & 0x80 == 0 { break;}
+                if (b as u64) & 0x80 == 0 {
+                    break;
+                }
                 offset += 7;
             }
             quantities.push(value);
         }
 
-        let payload = Payload { quantities, metadata: Decodable::consensus_decode(d)? };
+        let payload = Payload {
+            quantities,
+            metadata: Decodable::consensus_decode(d)?,
+        };
         return Ok(payload);
     }
 }
 
-pub trait TxOutExt{
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub struct Metadata(Vec<u8>);
+
+impl Metadata {
+    pub fn to_string(&self) -> Result<String, FromUtf8Error> {
+        String::from_utf8(self.0.clone())
+    }
```

Comment:
How about using ToString Trait?

Reply:
I've implemented fmt::Display trait, instead of ToString trait.

https://doc.rust-lang.org/std/string/trait.ToString.html
```
This trait is automatically implemented for any type which implements the Display trait. As such, ToString shouldn't be implemented directly: Display should be implemented instead, and you get the ToString implementation for free.
```